tests/unit/base/test_deprecation.py

```diff
@@ -0,0 +1,43 @@
+import unittest
+import warnings
+
+from twilio.base.obsolete import deprecated_method
+
+
+class DeprecatedMethodTest(unittest.TestCase):
+
+    def test_deprecation_decorator(self):
+
+        @deprecated_method()
```

Comment:
FYI: Python makes it non-trivial to have a decorator that works with and without `()`. I've don't it before, but it's not fun to figure out how to do it.

Reply:
Updated the definition using [this stackoverflow response](https://stackoverflow.com/a/35572746). Was actually fairly straightforward.

Reply:
Hrm, I think at the time I was making it generic so it would work on any decorator. Or it was a decorator that you could apply to other decorators.